dyno-contrib/src/main/java/com/netflix/dyno/contrib/consul/ConsulHostsSupplier.java

```diff
@@ -107,7 +107,7 @@ public Host apply(HealthService info) {
 
                         Host.Status status = Host.Status.Up;
                         for (com.ecwid.consul.v1.health.model.Check check : info.getChecks()) {
-                            if (check.getStatus().equals(Check.CheckStatus.CRITICAL)) {
+                            if (check.getStatus() == Check.CheckStatus.CRITICAL) {
```

Comment:
In general, `==` checks if both objects point to the same memory location whereas .equals() evaluates to the comparison of values in the objects. Are you sure this is a correct fix?

Reply:
Yes. The equals method in Enum.java does an `==` comparison underneath. http://hg.openjdk.java.net/jdk7/jdk7/jdk/file/ff9ad99b63cc/src/share/classes/java/lang/Enum.java#l134

In general to check for equality in enum, using `==` is a better option than the equals method since 
1) It will check the type of the enum compared are the same during compile time (Which would have prevented this bug) 
2) It will eliminate the possibility of NPE.

Reply:
@ipapapa Can you also release it to maven?